node/src/Pos/Wallet/Web/Util.hs

```diff
@@ -55,13 +56,24 @@ getWalletAddrMetas lookupMode cWalId =
 getWalletAddrs
     :: (WebWalletModeDB ctx m, MonadThrow m)
     => AddressLookupMode -> CId Wal -> m [CId Addr]
-getWalletAddrs = (cwamId <<$>>) ... getWalletAddrMetas
+getWalletAddrs mode wid = cwamId <<$>> getWalletAddrMetas mode wid
 
-getWalletAddrsSet
+getWalletAddrsDetector
     :: (WebWalletModeDB ctx m, MonadThrow m)
-    => AddressLookupMode -> CId Wal -> m (Set (CId Addr))
-getWalletAddrsSet lookupMode cWalId =
-    S.fromList . map cwamId <$> getWalletAddrMetas lookupMode cWalId
+    => AddressLookupMode -> CId Wal -> m (CId Addr -> Bool)
+getWalletAddrsDetector lookupMode cWalId = do
+    accIds <- getWalletAccountIds cWalId
+    case accIds of
+        [accId] -> do
+            -- can avoid conversion to Set
+            addrMaps <- getAccountAddrMaps accId
+            return $ \addr ->
+                any (HM.member addr) [getCurrent addrMaps, getRemoved addrMaps]
```

Comment:
This case does not respect the `lookupMode`. Either we should do the case above to special-case on `Ever` or we should handle the other two cases too. e.g.

    case (accIds, lookupMode) of
      ([accId], Ever) -> do

Overall this isn't wrong because we currently only call it with `Ever`, but it's just asking for trouble later.